public/static/sdk/jsmr-0.1.js

```diff
@@ -1,104 +1,155 @@
+                    return function () {
                         var tmp = this._super;
-                     
+
                         // Add a new ._super() method that is the same method
                         // but on the super-class
                         this._super = _super[name];
-                     
+
                         // The method only need to be bound temporarily, so we
                         // remove it when we're done executing
                         var ret = fn.apply(this, arguments);
                         this._super = tmp;
-                     
+
                         return ret;
                     };
                 })(name, prop[name]) :
                 prop[name];
         }
-     
+
         // The dummy class constructor
         function Class() {
             // All construction is actually done in the init method
-            if ( !initializing && this.init )
+            if (!initializing && this.init)
                 this.init.apply(this, arguments);
         }
-     
+
         // Populate our constructed prototype object
         Class.prototype = prototype;
-     
+
         // Enforce the constructor to be what we expect
         Class.prototype.constructor = Class;
- 
+
         // And make this class extendable
         Class.extend = arguments.callee;
-     
+
         return Class;
     };
 })();
 
+
+
 var JsMr = Class.extend({
     init: function () {
+
         this.registered = false;
         this.client_id = null;
         this.last_call = 0;
         this.current_task = null;
+        this.current_output = "";
+        this.current_state = null;
+        this.step_list = [];
+
         this.beat_interval_obj = null;
+        this.aws_sdk = 'https://sdk.amazonaws.com/js/aws-sdk-2.1.21.min.js';
         this.options = {
             beat_interval: 60000, // one minute
             auth_token: null,
             server_path: '',
             debug: true,
-            jquery_cdn: 'https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js',
-        }
+            jquery_cdn: 'https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js'
+        };
 
         // Start the SDK
         this.loadAndRegister();
     },
 
     /**
+     * Upload the output and state to AWS S3
+     *
+     */
+    upload_data: function () {
+
+        var s3 = new AWS.S3();
+        var task = this.current_task;
+        var job_id = task.job_id.toString();
+        var task_id = task.task_id.toString();
+        var state = JSON.stringify(this.current_state);
+
+        console.log('Output is' + this.current_output);
+        console.log('State is' + state);
+
+        var output_params = {
+            Bucket: task.bucket_name, /* required */
+            Key: job_id.concat('_', task.task_id, '_output'),
+            Body: this.current_output
+        };
+
+        var state_params = {
+            Bucket: task.bucket_name, /* required */
+            Key: job_id.concat('_state'),
```

Comment:
state needs the `step` and `instance` id in the key as well so that server can find it.